dima/react-movies/src/components/searchBar/searchBar.tsx

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import SearchFilter from "./searchFilter";
+import "./searchBar.css";
+import ErrorBoundary from "../errorBoundary";
+import * as QueryString from "query-string";
+import FilterProperty from "../../enums/FilterPropery";
+import { RouteComponentProps } from "react-router";
+import ParamsToPush from "../../interfaces/paramsToPush";
+interface SearchBarProps {
+  location: RouteComponentProps["location"];
+  pushParamsOnSubmit: (urlParams: ParamsToPush) => void;
+}
+
+interface SearchBarState {
+  searchBarValue: string | undefined;
+  searchBy: FilterProperty;
+}
+
+export default class SearchBar extends React.Component<
+  SearchBarProps,
+  SearchBarState
+> {
+  state: SearchBarState = {
+    searchBarValue: "",
+    searchBy: FilterProperty.title,
+  };
+
+  componentDidMount() {
+    if (QueryString.parse(this.props.location.search).searchBy) {
+      this.toggleSearchCategoryByURL();
+    }
+    const { search } = QueryString.parse(this.props.location.search);
+    if (search) {
+      this.setState({ searchBarValue: search.toString() });
+    }
```

Comment:
It is not really good approach to update state several times in a row. Move it to your **toggleSearchCategoryByURL** method.  And rename **toggleSearchCategoryByURL** to some setFiltersFromUrl or smth like that
```
this.setState({
       search: searchFromUrl || '' ;
      searchBy:
        QueryString.parse(this.props.location.search).searchBy ===
        FilterProperty.title
          ? FilterProperty.title
          : FilterProperty.genre,
    });
```